tools/publish_snapshots.main.kts

```diff
@@ -0,0 +1,277 @@
+@file:ScriptFileLocation("scriptPath")
+
+import java.io.BufferedReader
+import java.io.File
+import java.io.InputStreamReader
+import kotlin.system.exitProcess
+
+/**
+ * Script that will take a local maven repo and clone it to Maven Central.
+ * The script supports both normal and SNAPSHOT releases.
+ *
+ * This makes it possible to create a full maven repository across multiple
+ * GitHub Action runners and then finally upload all artifacts in one go once
+ * all tests have passed.
+ */
+// Wrapper describing a single file in a package + including metadata needed when uploading to Maven Central.
+data class FileDescriptor(val fileName: String, val classifier: String, val type: String)
+// Wrapper for data required to upload a single package to Maven Central.
+data class PackageData(
+    val fullPathToPackage: String,
+    val pomFile: FileDescriptor,
+    val mainFile: FileDescriptor,
+    val additionalFiles: List<FileDescriptor>)
+
+fun debug(message: String?) {
+    println(message)
+}
+
+if (args.size != 6) {
+    println("Usage: kotlin publish_snapshots.main.kts <path_to_kotlin_repo> <version> <gpg_signing_key> <gpg_pass_phrase> <maven_central_username> <maven_central_password>")
+    exitProcess(1)
+}
+
+// Constants
+// Only files of this type are candidates to be the _main_ file
+val mainFileTypes = listOf("aar", "jar", "klib")
+// Files with this type is ignored when determining which files to upload
+val ignoredFileTypes = listOf("md5", "sha1", "sha256", "sha512", "asc")
+// Files that match this name is ignored when determining which files to upload
+val ignoreFiles = listOf("maven-metadata.xml")
+// Full path to the root of the realm-kotlin repo
+val repoPath: String = File(args[0]).absolutePath
+// Version of the SDK to upload, this can both be a full release or a -SNAPSHOT release
+val version = args[1]
+// SNAPSHOT releases are only marked as such in the folder structure, files inside contain only the version without
+// the -SNAPSHOT suffix.
+val versionPrefix = version.removeSuffix("-SNAPSHOT")
+// Wether or not we are about to upload a SNAPSHOT release.
+val isSnapshot = version.endsWith("-SNAPSHOT", ignoreCase = true)
+// Secret key used to sign artifacts. Must be encoded using base64.
+// The following code can be used to create this:
+// # Export base64 key
+// gpg --export-secret-key --armor | base64
+// # Import base64 key with a passphrase
+// echo $BASE64_SIGNING_KEY | base64 -d | gpg --batch --import
+val gpgBase64SigningKey = args[2]
+val gpgPassPhrase = args[3]
+// Maven Central username and password token.
+// Can be found by logging in to https://oss.sonatype.org/#welcome and go to Profile > User Token.
+val sonatypeUsername = args[4]
+val sonatypePassword = args[5]
+// Path to the root of the local m2 repo containing the package to release.
+val localMavenRepo = File("$repoPath/packages/build/m2-buildrepo")
```

Comment:
Should we consider to somehow hook this up with https://github.com/realm/realm-kotlin/blob/main/packages/gradle.properties#L35